testing/ef-tests/tests/evm.rs

```diff
@@ -55,57 +48,26 @@ fn evm(
             break;
         }
 
-        let block_header = block.block_header.unwrap();
-        let expected_header: Header = block_header.clone().into();
-        assert_eq!(&expected_header.hash(), &block_header.hash);
+        let expected_header = block.block_header.as_ref().unwrap();
+        let expected_hash = expected_header.hash;
 
         // using the empty/default state for the input prepares all accounts for deletion
         // this leads to larger input, but can never fail
-        let post_state = json.post.clone().unwrap_or_default();
-
-        let input = create_input(
-            &chain_spec,
-            genesis,
-            json.pre,
-            expected_header.clone(),
-            block.transactions,
-            block.withdrawals.unwrap_or_default(),
-            post_state,
-        );
-        let input_state_input_hash = input.state_input.hash();
+        let post_state = json.post_state.clone().unwrap_or_default();
 
-        let output = EthereumStrategy::build_from(&chain_spec, input).unwrap();
+        let provider = TestProvider::new(json.genesis_block_header, block, json.pre, post_state);
+        let provider = Rc::new(RefCell::new(provider));
 
-        let BlockBuildOutput::SUCCESS {
-            hash: new_block_hash,
-            head: new_block_head,
-            state: new_block_state,
-            state_input_hash,
-        } = output
-        else {
-            panic!("Invalid block")
-        };
-
-        if let Some(post) = json.post {
-            let (exp_state, _) = ethers::build_tries(&post);
-
-            println!("diffing state trie:");
-            for diff in diff::slice(
-                &new_block_state.debug_rlp::<StateAccount>(),
-                &exp_state.debug_rlp::<StateAccount>(),
-            ) {
-                match diff {
-                    diff::Result::Left(l) => println!("✗{}", l),
-                    diff::Result::Right(r) => println!("✓{}", r),
-                    diff::Result::Both(l, _) => println!(" {}", l),
-                }
-            }
-            assert_eq!(new_block_state.hash(), exp_state.hash());
-        }
+        let preflight_data =
+            <<RethBlockBuilder as BlockBuilder<_, _, TestCoreDriver, _>>::PreflightClient as PreflightClient<
+                Ethereum,
+                TestCoreDriver,
+                RethPreflightDriver,
+            >>::preflight_with_provider(provider, 1, 1)
+                .unwrap();
+        let build_result = Witness::driver_from::<TestCoreDriver>(&preflight_data);
```

Comment:
We need a `StatelessClient::validate` call here

Reply:
Addressed in 465de7927675d0e9d71794844c6d903d0fe0d483